homeassistant/components/zwave_js/sensor.py

```diff
@@ -626,11 +627,19 @@ class ZWaveNumericSensor(ZwaveSensor):
     """Representation of a Z-Wave Numeric sensor."""
 
     @callback
-    def on_value_update(self) -> None:
+    def on_value_update(self, event_data: dict) -> None:
         """Handle scale changes for this value on value updated event."""
-        data = NumericSensorDataTemplate().resolve_data(self.info.primary_value)
+        value = self.info.primary_value
+        data = NumericSensorDataTemplate().resolve_data(value)
         self.entity_description = get_entity_description(data)
         self._attr_native_unit_of_measurement = data.unit_of_measurement
+        if (
+            (prev_value := event_data["args"]["prevValue"]) is not None
+            and (new_value := event_data["args"]["newValue"]) is not None
+            and prev_value > new_value
+            and self.state_class == SensorStateClass.TOTAL
```

Comment:
Why do we use this kind of check? Don't we know the time of last reset since it's called "today"? Otherwise maybe we should just use state class `TOTAL_INCREASING` instead?

Reply:
Today is relative to what the device thinks right? I have no way of knowing what today represents for the device at the moment. I will switch to `TOTAL_INCREASING` instead

Reply:
If the device collects data for a certain time period I'd expect it to either report the time period or define the time criteria so we can keep track of time ourselves.

Reply:
Understood, perhaps we can switch back to total later when we have a better understanding of how to get time. It would also be nice to get a device dump for a device that actually has this CC